be/src/vec/runtime/vdatetime_value.h

```diff
@@ -288,21 +288,23 @@ class VecDateTimeValue { // Now this type is a temp solution with little changes
         case YEAR: {
             int year = (ts_value2.year() - ts_value1.year());
             if (year > 0) {
-                year -= (ts_value2.to_int64() % 10000000000 - ts_value1.to_int64() % 10000000000) <
-                        0;
+                year -= (ts_value2.to_datetime_int64() % 10000000000 -
```

Comment:
Why not use to_int64 here? I find it will check the type of datetimevalue. 
int64_t VecDateTimeValue::to_int64() const {
    switch (_type) {
    case TIME_TIME:
        return to_time_int64();
    case TIME_DATE:
        return to_date_int64();
    case TIME_DATETIME:
        return to_datetime_int64();
    default:
        return 0;
    }
}
Are you sure, that the value here is always with TIME_DATETIME type?

Reply:
Because the logic of `% 10000000000` is depend on `datetime` type.
If the type is `date`, this logic should be like this `(ts_value2.to_int64() % 10000 - ts_value1.to_int64() % 10000) < 0`

Reply:
Sorry for late reply. I will merge it. Thanks for your contribution.